app/components/Taps/TapConfiguration/S3/index.js

```diff
@@ -0,0 +1,350 @@
+                <FormFeedback>Required</FormFeedback>
+              </FormGroup>
+            </Col>
+          </Row>
+          <Row>
+            <Col>
+              <FormGroup>
+                <Label for="start_date">Start date</Label>
+                <Input
+                  type="date"
+                  name="start_date"
+                  id="start_date"
+                  value={start_date ? this.formatDate(start_date) : ''}
+                  onBlur={(event) => {
+                    const { value } = event.currentTarget;
+                    this.validate('start_date', value);
+                  }}
+                  onChange={this.handleChange}
+                  {...this.state.start_date}
+                />
+                <FormText>
+                  Applies to tables with a defined timestamp field and limits
+                  how much historical data will be replicated.
+                </FormText>
+                <FormFeedback>Required</FormFeedback>
+              </FormGroup>
+            </Col>
+          </Row>
+          <Row>
+            <Table className="table-borderless">
+              <thead>
+                <tr>
+                  <th className="font-weight-normal">Table name</th>
+                  <th className="font-weight-normal">S3 key pattern</th>
+                  <th className="font-weight-normal">Key field(s)</th>
+                </tr>
+              </thead>
+              <tbody>
+                {this.state.tables.map((table, idx) => (
+                  <tr>
+                    <td>
+                      <Input
+                        type="text"
+                        name="table_name"
+                        id="table_name"
+                        value={table.table_name}
+                        onChange={this.handleTableChange(idx)}
+                        {...this.state.tables.table_name}
+                      />
+                    </td>
+                    <td>
+                      <Input
+                        type="text"
+                        name="search_pattern"
+                        id="search_pattern"
+                        value={table.search_pattern}
+                        onChange={this.handleSearchPatternChange(idx)}
+                        {...this.state.tables.search_pattern}
+                      />
+                    </td>
+                    <td>
+                      <ReactTags
+                        name="key_properties"
+                        id="key_properties"
+                        tags={this.state.tags}
+                        handleAddition={this.handleKeyFieldChange(idx)}
+                        handleDelete={this.handleDelete}
+                        delimiters={delimiters}
+                        placeholder="Add new key field"
+                        autofocus={false}
+                        classNames={{
+                          tag: styles.inputTagValue,
+                          tagInputField: 'form-control',
```

Comment:
Change 'form-control' to 'form-control formcontrol-sm'

Reply:
I added 'form-control form-control-sm' and the input field looks weird not sure if that was the intention.

![image](https://user-images.githubusercontent.com/9654881/43641975-8e778bf2-971d-11e8-905a-283629454236.png)